var/spack/repos/builtin/packages/raja/package.py

```diff
@@ -409,46 +409,52 @@ def setup_build_tests(self):
+        mkdirp(install_test_root(self))
 
     @property
     def _extra_tests_path(self):
         # TODO: The tests should be converted to re-build and run examples
         # TODO: using the installed libraries.
-        return join_path(self.install_test_root, self.build_relpath, "bin")
-
-    def _test_examples(self):
-        """Perform very basic checks on a subset of copied examples."""
-        checks = [
-            (
-                "ex5_line-of-sight_solution",
-                [r"RAJA sequential", r"RAJA OpenMP", r"result -- PASS"],
-            ),
-            (
-                "ex6_stencil-offset-layout_solution",
-                [r"RAJA Views \(permuted\)", r"result -- PASS"],
-            ),
-            (
-                "ex8_tiled-matrix-transpose_solution",
-                [r"parallel top inner loop", r"collapsed inner loops", r"result -- PASS"],
-            ),
-            ("kernel-dynamic-tile", [r"Running index", r"(24,24)"]),
-            ("plugin-example", [r"Launching host kernel for the 10 time"]),
-            ("tut_batched-matrix-multiply", [r"result -- PASS"]),
-            ("wave-eqn", [r"Max Error = 2", r"Evolved solution to time"]),
-        ]
-        for exe, expected in checks:
-            reason = "test: checking output of {0} for {1}".format(exe, expected)
-            self.run_test(
-                exe,
-                [],
-                expected,
-                installed=False,
-                purpose=reason,
-                skip_missing=True,
-                work_dir=self._extra_tests_path,
-            )
+        return join_path(install_test_root(self), self.build_relpath, "bin")
+
+    def run_raja(self, exe, expected):
+        example_exe = which(exe)
+        if example_exe is None:
+            raise SkipTest(f"{exe} is not installed for version {self.version}")
+        out = example_exe([], output=str.split, error=str.split)
+        check_outputs(expected, out)
+
+    def test_sequential(self):
+        """Test raja sequential"""
+        self.run_raja(
+            "ex6_stencil-offset-layout_solution", [r"RAJA Views \(permuted\)", r"result -- PASS"]
+        )
+
+    def test_views(self):
+        """Test raja views"""
+        self.run_raja(
+            "ex6_stencil-offset-layout_solution", [r"RAJA Views \(permuted\)", r"result -- PASS"]
+        )
+
+    def test_matrix(self):
+        """Test raja matrix"""
+        self.run_raja(
+            "ex8_tiled-matrix-transpose_solution",
+            [r"parallel top inner loop", r"collapsed inner loops", r"result -- PASS"],
+        )
+
+    def test_dynamic(self):
+        """Test dynamic kernel"""
+        self.run_raja("kernel-dynamic-tile", [r"Running index", r"(24,24)"])
+
+    def test_plugin_example(self):
+        """Test plugin example"""
+        self.run_raja("plugin-example", [r"Launching host kernel for the 10 time"])
+
+    def test_tut_matrix(self):
+        """Test tut matrix"""
```

Comment:
The name and docstring/purpose aren't very descriptive.  It's really batched matrix multiple.